integrated_channels/canvas/client.py

```diff
@@ -90,9 +109,36 @@ def _delete(self, url, data):
             data (str): The json encoded payload to DELETE.
         """
         self._create_session()
-        response = self.session.delete(url, data=data)
+
+        integration_id = json.loads(
+            data.decode("utf-8")
+        )['course']['integration_id']
+        course_id = self._get_course_id_from_integration_id(integration_id)
+
+        url = '{}/api/v1/courses/{}'.format(
+            self.enterprise_configuration.canvas_base_url,
+            course_id,
+        )
+        response = self.session.delete(url, data='{"event":"delete"}')
         return response.status_code, response.text
 
+    def _get_course_id_from_integration_id(self, integration_id):
+        """
+        doc TBD
+        """
+        url = "{}/api/v1/accounts/{}/courses/".format(
+            self.enterprise_configuration.canvas_base_url,
+            self.enterprise_configuration.canvas_account_id
+        )
+        all_courses_response = self.session.get(url).json()
+
+        for course in all_courses_response:
+            if course['integration_id'] == integration_id:
+                # raise if we find two matching?
+                # break if we find a matching integration id?
+                course_id = course['id']
```

Comment:
hmm, wouldn't we want to end the loop once we find the course_id? ... unless we want to find duplicates etc which I think is not needed since Canvas already controls uniqueness of integration_id, I am pretty sure. But please do confirm that .. if we do then we can early return rather than loop the rest of the items

Reply:
Yup, I wasn't sure what the possibilities were in terms of if we could get dups, so initially I added the check. I am now more sure that it will never happen so I added a break 👍